### PR TITLE
Add .editorconfig file

### PR DESCRIPTION
**Pull request by mloskot**
Propose basic settings for controlling EOL encoding and space-only indentation of CMake and other scripts.

## Tasklist

 - [x] review
 - [x] adjust for comments

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# EditorConfig is awesome: http://EditorConfig.org
+#
+# NOTE: Keep settings in sync with the master .clang-format file
```
Comment (TheMarex) [inline .editorconfig]: I don't see rules for the `.{h,c}pp` files, is that something that is not usually captured in this config?

Comment (mloskot) [reply .editorconfig]: @TheMarex This config can control encoding/indentation of any text file (see an example https://github.com/nanodbc/nanodbc/blob/master/.editorconfig).

I just didn't know why `.editorconfig` is not used by the project, perhaps that was deliberate decision. I did not want to be too intrusive with rules too many rules. I'm also not sure waht should be the value of `insert_final_newline`, `trim_trailing_whitespace`, etc. for OSRM codebase.

So, I decided to utilise `.editorconfig` for the files which are not handled by `clang-format`. By the way, `.editorconfig` is not a replacement for `clang-format` - it is supporting it during typing, `clang-format` acts a bit later :)

Comment (mloskot) [reply .editorconfig]: @TheMarex Do you wish me to update `.editorconfig` for C++ and other files?

Comment (TheMarex) [reply .editorconfig]: @mloskot no I think this approach is fine. Could you try to re-trigger the travis builds with a new commit? Seems like it wasn't picked up.

Comment (mloskot) [reply .editorconfig]: @TheMarex OK, commit update pushed

**Pull request merged**